### DISCUSSION TITLE
Next.js does not recognize dynamic routes based on filesystem

### DISCUSSION THREAD

**Discussion by abdulsalamalmahdi** [vercel/next.js#35544] [Help] [answered]
![image](https://user-images.githubusercontent.com/35931165/159690131-6be021b0-b48a-4f6b-8472-72bf2e77e15a.png)

 - pages 
   - api
     - products 
       - pro
         - [id].js
     - add.js
[id].js

import { upload, baseHandler } from 'helpers';
import { sequelize } from 'lib';
import { init, Suppliers, Products} from 'models';
import { supplierController, productController } from 'controllers';

const handler = baseHandler.get( async (req, res) => {
    const {
        query: { id, name },
        method,
      } = req

    init()

    res.status(400).json(id)
})
 
export default handler;

export const config = {
  api: {
    bodyParser: false,
  },
};
The static routes works fine but not the dynamic ones. I get this error when I try to access them error - Error: getaddrinfo ENOTFOUND api.example.com which means Next.js does not add dynamic routes from the file system.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Well it does look like `api.example.com` does not exist, how are you making your requests?

What happens if you remove `init` and unwrap the functions from `baseHandler`, then everything works right? We may need to take a look inside `baseHandler`, or `init` to see why/how it ends up calling an non existing URL.

**Comment by abdulsalamalmahdi** [comment]
for the sake of it I changed the code inside [id].js to the snippet from the docs 

export default function handler(req, res) {
  const { id } = req.query
  res.end(`Post: ${id}`)
}
  still got that one line error

**Reply by icyJoseph** [reply]
How are you calling the endpoint?

**Reply by abdulsalamalmahdi** [reply]
I am calling it through postman  http://localhost:5000/api/products/pro/123

**Reply by icyJoseph** [reply]
Ah, strange, I just tested this and it works, no problem. Do you have a link to a repository? or perhaps there's some configuration missing? Are you using `next dev` or `next start`?

**Reply by abdulsalamalmahdi** [reply]
this is the repository link https://github.com/abdulsalamalmahdi/upload

I am using next dev

tried to use custom server but that kind of defeat the purpose.

**Reply by icyJoseph** [reply]
What's up with this rewrite? This is the problem. What are you trying to achieve? Moving `/api/:path*` requests to be reachable from `/:path*` directly?

```js
 async rewrites() {
      return [
        {
          source: '/api/:path*',
          destination: 'https://api.example.com/:path*',
        },
      ]
    },
```

**Reply by abdulsalamalmahdi** [reply]
thans a lot, I just forgot that I added it to  use the dynamic routing provided by Next.js, and I did not want to use the necessary file structure 
then somewhy I changed my mind and forget to remove the rewrites function.
appreciate it

**Reply by icyJoseph** [reply]
Awesome! Don't forget to mark as answered, I also learned about `next-connect` thanks for that.

**Reply by abdulsalamalmahdi** [reply]
done. yes glad I learnt something in return. NC is great
